homeassistant/components/lcn/climate.py

```diff
@@ -119,10 +119,14 @@ async def async_set_hvac_mode(self, hvac_mode):
         """Set new target hvac mode."""
         if hvac_mode == const.HVAC_MODE_HEAT:
             self._is_on = True
-            self.address_connection.lock_regulator(self.regulator_id, False)
+            self.hass.async_create_task(
```

Comment:
Why do we not await the coroutine?

Reply:
When sending the command to the hardware device an acknowledge is requested. If no acknowledge is received after a certain time (usually after 3.5s) the command is resend up to three times. This might happen due to heavy communication traffic.
In the worst (rare) case we await here up to >10s before HA gets updated. That‘s why I thought wrapping everything in tasks would be the better solution.

Reply:
When the acknowledge comes in, the state is updated to the correct state?

Reply:
I did some investigations here and maybe found a better solution.
If we await the coroutines (e.g. `lock_regulator), they return a boolean indicating if the acknowledge was received or not. We could check that and if there was no acknowledge received, we could skip changing HA states because we can be sure that the hardware device never changed its state. Works like a charm.

But there is one drawback: When changing many devices at the same time and one device does not respond, we get a *blocking* beahvior for >10s until the corresponding command fails. Only after that the other devices get their command. We can overcome this by setting the `PARRALLEL_UPDATES = 0` for the platforms.
If that's ok for you, I would prefer this solution as it feels somehow *better*.

Reply:
It would be like this:
```python
if not await self.address_connection.lock_regulator(self.regulator_id, False):
    return
self._is_on = True
self.async_write_ha_state()
```

Reply:
Does the hardware handle the parallel service calls well?

Reply:
Yes, LCN is basically a bus system connected via a TCP/IP socket to the host. So there are plenty of buffers for sending/receiving messages.
In addition, one of the most important things the underlying library has to do is to ensure correct concurrent message handling.

Reply:
Great!